utils/script-luajit.c

```diff
@@ -0,0 +1,425 @@
+			switch (spec->size) {
+			case 1:
+				dllua_pushinteger(L, 1 + count++);
+				dllua_pushinteger(L, val.c);
+				dllua_settable(L, -3);
+				break;
+			case 2:
+				dllua_pushinteger(L, 1 + count++);
+				dllua_pushinteger(L, val.s);
+				dllua_settable(L, -3);
+				break;
+			case 4:
+				dllua_pushinteger(L, 1 + count++);
+				dllua_pushinteger(L, val.i);
+				dllua_settable(L, -3);
+				break;
+			case 8:
+				dllua_pushinteger(L, 1 + count++);
+				dllua_pushinteger(L, val.L);
+				dllua_settable(L, -3);
+				break;
+			default:
+				pr_warn("invalid argument format: %d\n", spec->fmt);
+				break;
+			}
+			data += ALIGN(spec->size, 4);
+			break;
+		case ARG_FMT_FLOAT:
+			memcpy(val.v, data, spec->size);
+			switch (spec->size) {
+			case 4:
+				dval = val.f;
+				break;
+			case 8:
+				dval = val.d;
+				break;
+			case 10:
+				dval = (double)val.D;
+				break;
+			default:
+				pr_dbg("invalid floating-point type size %d\n",
+				       spec->size);
+				dval = 0;
+				break;
+			}
+			dllua_pushinteger(L, 1 + count++);
+			dllua_pushnumber(L, dval);
+			dllua_settable(L, -3);
+			data += ALIGN(spec->size, 4);
+			break;
+
+		case ARG_FMT_STR:
+		case ARG_FMT_STD_STRING:
+			/* get string length (2 bytes in the beginning) */
+			memcpy(&slen, data, 2);
+
+			str = xmalloc(slen + 1);
+
+			/* copy real string contents */
+			memcpy(str, data + 2, slen);
+			str[slen] = '\0';
+
+			/* NULL string is encoded as '0xffffffff' */
+			if (!memcmp(str, &null_str, sizeof(null_str)))
+				strcpy(str, "NULL");
+
+			dllua_pushinteger(L, 1 + count++);
+			dllua_pushstring(L, str);
+			dllua_settable(L, -3);
+			free(str);
+			data += ALIGN(slen + 2, 4);
+			break;
+
+		case ARG_FMT_CHAR:
+			/* make it a string */
+			memcpy(ch_str, data, 1);
+			ch_str[1] = '\0';
+			dllua_pushinteger(L, 1 + count++);
+			dllua_pushstring(L, ch_str);
+			dllua_settable(L, -3);
+			data += 4;
+			break;
+
+		default:
+			break;
+		}
+	}
+	if (is_retval) { // * ctx 'retval' retval
+		dllua_pushinteger(L, 1); // * ctx 'retval' retval-table 1
+		dllua_gettable(L, -2); // * ctx 'retval' retval-table retval
+		dllua_remove(L, -2); // * ctx 'retval' retval
+	}
+	dllua_settable(L, -3); // * ctx
+}
+
+static int luajit_uftrace_begin(struct script_info *info)
+{
+	pr_dbg("%s()\n", __func__);
+
+	lua_getglobal(L, "uftrace_begin"); // uftrace_begin
+	if (lua_isnil(L, -1)) // nil
+	{
+		lua_pop(L, 1); //
+		return -1;
+	}
+
+	lua_newtable(L); // uftrace_begin ctx
+	dllua_pushstring(L, "record"); // uftrace_begin ctx 'record'
+	dllua_pushboolean(L, info->record); // uftrace_begin ctx 'record' record
+	dllua_settable(L, -3); // uftrace_begin cx
+	dllua_pushstring(L, "version"); // uftrace_begin ctx 'version'
+	dllua_pushstring(L, info->version); // uftrace_begin ctx 'version' version
+	dllua_settable(L, -3); // uftrace_begin ctx
+	dllua_pushstring(L, "cmds"); // uftrace_begin ctx 'cmds'
+	lua_newtable(L); // uftrace_begin ctx 'cmds' cmds
+	int i;
+	char *s;
```

Comment:
I know it's old-fashioned, but I prefer putting declarations at the beginning of function.  Or at least you can use a blank line to separate them from others.

Reply:
Fixed